lib/pbench/agent/tool_meister.py

```diff
@@ -1316,121 +1316,120 @@ def _send_directory(self, directory, uri, ctx):
         target_dir = directory.name
         parent_dir = directory.parent
         tar_file = parent_dir / f"{target_dir}.tar.xz"
-        o_file = parent_dir / f"{target_dir}.tar.out"
-        e_file = parent_dir / f"{target_dir}.tar.err"
+        tar_args = [
+            self.tar_path,
+            "--create",
+            "--xz",
+            "--force-local",
+            f"--file={tar_file}",
+            target_dir,
+        ]
         try:
             # Invoke tar directly for efficiency.
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
-                )
+            cp = subprocess.run(
+                tar_args,
+                cwd=parent_dir,
+                stdin=None,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
         except Exception:
             self.logger.exception("Failed to create tar ball '%s'", tar_file)
             failures += 1
         else:
             try:
                 if cp.returncode != 0:
-                    self.logger.error(
-                        "Failed to create tar ball; return code: %d", cp.returncode,
+                    self.logger.warning(
+                        "Failed to create tar ball first time; Error: %s, re-trying with --warning=none",
+                        str(cp.stdout),
                     )
-                    failures += 1
-                else:
+                    # record the tar error we encountered first time
+                    with open(target_dir / "tar_command_error") as file:
+                        file.write(str(cp.stdout))
```

Comment:
Or `(target_dir / "tar_command_error").write_text(str(cp.stdout))`??

Reply:
Or why not use the Redis logger?

Reply:
The more I think about it, there is nothing that will consume this file in the tar ball to take a different action.  For now having the warning log message above is sufficient.